# Survey of Consumer Finances (SCF)  
This notebook is going to be the first notebook in exploring the summarized version of the Survey of Consumer Finances. The summarized version is found on the website: https://www.federalreserve.gov/econres/scfindex.htm . We have decided to explore the summarized version first, as it has been cleaned and summarized by the Fed. They have actually created five different sets of the dataset with differing imputed values for missing values. We will need to pick one of the versions or potentially average the results from all five. Let's start by importing our libraries and downloading the dataset if don't yet have it.

## Libraries


In [ ]:
import requests
import zipfile
import io

import numpy as np
import pandas as pd
import altair as alt

# Deactivate the max rows and columns limit for Altair
alt.data_transformers.disable_max_rows()

url = "https://www.federalreserve.gov/econres/files/scfp2022excel.zip"

## Data Download or Import
Alright, now that we have all of this, I want to see if we have the file. If we have the file, we can just import it, however if we haven't downloaded it yet, we will go to the website and download it. 

In [ ]:
# Check to see if the data file already exists before attempting to download
try:
    print("Importing data from the Survey of Consumer Finances...")
    data = pd.read_csv("data/SCFP2022.csv")
    print("Data imported successfully.")

#If it doesn't exist we can download it form the Federal Reserve's website
except:
    print("File not found. Downloading survey data from the Federal Reserve's website...")
    headers = {"User-Agent": "Mozilla/5.0"}
    response = requests.get(url, headers=headers)
    print(f"Status code: {response.status_code}") # Check if the request was successful
    if response.status_code == 200:
        with zipfile.ZipFile(io.BytesIO(response.content)) as z:
            z.extractall("data")  # Extracts all files in the zip to the current directory
        print("Data downloaded and extracted successfully.")
        data = pd.read_csv("data/SCFP2022.csv")
    else:
        print("Failed to download data. Please check the URL and try again.")


Importing data from the Survey of Consumer Finances...
Data imported successfully.


## Inital Exploration
To start, we should just get a sense of the amount of data that we have in our dataframe. What missing values we have, the columns, we have, descriptive statistics etc. 

In [12]:
data.shape

(22975, 357)

Alright, 357 columns is going to be way more than we need. So let's take a look at the documentation on the website to see which ones will likely be useful and which likely won't be. 

In [13]:
data.describe()

,YY1,Y1,WGT,HHSEX,AGE,AGECL,EDUC,EDCL,MARRIED,KIDS,...,NWCAT,INCCAT,ASSETCAT,NINCCAT,NINC2CAT,NWPCTLECAT,INCPCTLECAT,NINCPCTLECAT,INCQRTCAT,NINCQRTCAT
count,22975.000000,22975.000000,22975.000000,22975.000000,22975.000000,22975.000000,22975.000000,22975.000000,22975.000000,22975.000000,...,22975.000000,22975.000000,22975.000000,22975.000000,22975.000000,22975.000000,22975.000000,22975.000000,22975.000000,22975.000000
mean,2302.597171,23028.971708,5715.185610,1.238955,54.468988,3.498368,10.328400,3.111164,1.367791,0.738629,...,3.055321,3.620022,3.647095,3.651273,1.859935,6.670729,6.603395,6.671425,2.771665,2.792644
std,1328.749586,13287.495936,4966.430219,0.426455,16.189081,1.556147,2.815743,1.026125,0.482215,1.108022,...,1.528703,1.829824,1.880664,1.822568,0.803743,3.669262,3.573612,3.560253,1.187083,1.180095
min,1.000000,11.000000,2.754882,1.000000,18.000000,1.000000,-1.000000,1.000000,1.000000,0.000000,...,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000
25%,1152.000000,11524.500000,1850.200822,1.000000,42.000000,2.000000,8.000000,2.000000,1.000000,0.000000,...,2.000000,2.000000,2.000000,2.000000,1.000000,3.000000,3.000000,4.000000,2.000000,2.000000
50%,2303.000000,23033.000000,4083.825044,1.000000,56.000000,4.000000,11.000000,3.000000,1.000000,0.000000,...,3.000000,4.000000,4.000000,4.000000,2.000000,7.000000,7.000000,7.000000,3.000000,3.000000
75%,3454.000000,34541.500000,8746.732248,1.000000,67.000000,5.000000,12.000000,4.000000,2.000000,1.000000,...,5.000000,6.000000,6.000000,6.000000,3.000000,10.000000,10.000000,10.000000,4.000000,4.000000
max,4603.000000,46035.000000,28265.283116,2.000000,95.000000,6.000000,14.000000,4.000000,2.000000,10.000000,...,5.000000,6.000000,6.000000,6.000000,3.000000,12.000000,12.000000,12.000000,4.000000,4.000000
